src/main/java/org/spongepowered/common/mixin/core/server/management/MixinPlayerInteractionManager.java

```diff
@@ -113,8 +115,21 @@ public EnumActionResult processRightClickBlock(EntityPlayer player, net.minecraf
         // Sponge Start - Create an interact block event before something happens.
         @Nullable final ItemStack oldStack = stack.copy();
 
+        Cause cause;
+        // Handle fake players
+        if (player instanceof FakePlayer) {
+            Optional<Object> simulated = ((IMixinWorldServer) player.world).getCauseTracker().getCurrentContext()
+                    .firstNamed(NamedCause.PLAYER_SIMULATED, Object.class);
+            if (simulated.isPresent()) {
+                cause = Cause.of(NamedCause.of(NamedCause.PLAYER_SIMULATED, simulated.get()));
+            } else {
+                cause = Cause.of(NamedCause.of(NamedCause.PLAYER_SIMULATED, ((EntityPlayer) player).getGameProfile()));
```

Comment:
It's probably unlikely, but safer than doing Optional.get() on something.
If it's a fake player from a forge mod, the cause tracker probably wont have the necessary objects in the context